src/connection.c

```diff
@@ -503,6 +508,52 @@ free_resolv_cb_data(struct resolv_cb_data *cb_data) {
 }
 
 static void
+send_proxy_header(int clientfd, const struct Connection *con) {
```

Comment:
Is it? `con->client.watcher.fd` is not yet set at this point, as the watcher has not yet been initialized.

Maybe moving it right before `ev_io_start` would work? I had never used libev before so seeing the `ev_io_init` had a reference to `connection_cb` I wasn't sure if this callback could have called at this point and potentially written data to the client buffer before the PROXY header.